src/services/dashboard-service.js

```diff
@@ -0,0 +1,140 @@
+import by from 'thenby';
+import RoomStore from '../stores/room-store.js';
+import UserStore from '../stores/user-store.js';
+import LessonStore from '../stores/lesson-store.js';
+import DocumentStore from '../stores/document-store.js';
+import { FAVORITE_TYPE, USER_ACTIVITY_TYPE } from '../domain/constants.js';
+
+class DashboardService {
+  static get inject() { return [UserStore, DocumentStore, RoomStore, LessonStore]; }
+
+  constructor(userStore, documentStore, roomStore, lessonStore) {
+    this.userStore = userStore;
+    this.roomStore = roomStore;
+    this.lessonStore = lessonStore;
+    this.documentStore = documentStore;
+  }
+
+  async getUserActivities({ userId, limit = 0 }) {
+    const user = await this.userStore.getUserById(userId);
+
+    const createdRooms = await this.roomStore.getRoomsCreatedByUser(userId);
+    const joinedRooms = await this.roomStore.getRoomsJoinedByUser(userId);
+    const updatedRooms = await this.roomStore.getRoomsUpdatedByUser(userId);
+    const createdDocuments = await this.documentStore.getDocumentsMetadataByCreatedBy(userId);
+    const updatedDocuments = await this.documentStore.getDocumentsMetadataByUpdatedBy(userId);
+    const lessons = await this.lessonStore.getLessonsMetadataByCreatedBy(userId);
```

Comment:
A second consideration: We query the complete db without any limits here multiple times as part of the browser requesting the page. I think there should be a limit in the db queries - and - even more important - an index on the exact fields used for the query on each collection.

Reply:
We could/should indeed limit each collection query to the provided `limit` (or a default value), that's true. However we'd need to introduces sorting too, as for these case we specifically need certain date fields sorted "desc". I'll make changes in this direction and see what I encounter.

Reply:
Done, tested and added indices, thanks for the review and please check again.